spec/adama/invoker_spec.rb

```diff
@@ -36,26 +54,75 @@
       allow(instance_3).to receive(:run)
       allow(instance_4).to receive(:run)
 
-      Invoker.invoke(*command_list)
+      invokable.invoke(*command_list)
+      invoke
     end
 
-    it 'calls .new on the commands in order' do
-      Invoker.call(**kwargs)
-      expect(Command1).to have_received(:new).ordered
-      expect(Command2).to have_received(:new).ordered
-      expect(Command3).to have_received(:new).ordered
-      expect(Command4).to have_received(:new).ordered
+    shared_examples 'calls the commands in order' do
+      it 'calls .new on the commands in order' do
+        expect(Command1).to have_received(:new).ordered
+        expect(Command2).to have_received(:new).ordered
+        expect(Command3).to have_received(:new).ordered
+        expect(Command4).to have_received(:new).ordered
+      end
 
-      kwargs.each do |key, value|
+      it 'calls .call on the commands in order' do
+        expect(instance_1).to have_received(:run).ordered
+        expect(instance_2).to have_received(:run).ordered
+        expect(instance_3).to have_received(:run).ordered
+        expect(instance_4).to have_received(:run).ordered
       end
     end
 
-    it 'calls .call on the commands in order' do
-      Invoker.call(**kwargs)
-      expect(instance_1).to have_received(:run).ordered
-      expect(instance_2).to have_received(:run).ordered
-      expect(instance_3).to have_received(:run).ordered
-      expect(instance_4).to have_received(:run).ordered
+    # There are two methods of passing an Invoker a list
+    # of commands to run.
+    #
+    # The first methods sets up the command list in the class
+    # definition, which stores the array of commands as a
+    # class member var. This means that every Invoker of that
+    # type calls the same comnmands.
+    #
+    # The second method is to initialize a new instance of
+    # an invoker and pass the list of commands to invoke
+    # to the instance. This means that we can dynamically
+    # change the command list per Invoker instance.
+
+    context 'When an Invoker class is passed a command list' do
+      # The first method:
+      #
+      #   class Invoker
+      #     include Adama::Invoker
+      #     invoke(Command1, Command2)
+      #   end
+      #
+      #   Invoker.call(foo: bar)
+      #
+      let(:invokable) { Invoker }
+      let(:invoke) { invokable.call **kwargs }
+
+      it_behaves_like 'calls the commands in order'
+    end
+
+    context 'When an Invoker instance is passed a command list' do
+      # The second method:
+      #
+      #   class Invoker
+      #     include Adama::Invoker
+      #   end
+      #
+      #   Invoker.new(foo: bar).invoke(Command1, Command2).run
```

Comment:
We should throw these examples into the README somewhere 👌

Reply:
Agreed, will do!